src/config.ts

```diff
@@ -0,0 +1,102 @@
+import path from 'path';
+import {cosmiconfigSync} from 'cosmiconfig';
+import {validate} from 'jsonschema';
+import {merge} from 'lodash';
+
+export interface SnowpackConfig {
+  dedupe?: string[];
+  namedExports?: {[filepath: string]: string[]};
+  options: {
+    babel?: boolean;
+    clean?: boolean;
+    dest?: string;
+    exclude?: string[];
+    externalPackage?: string[];
+    include?: string;
+    nomodule?: string;
+    nomoduleOutput?: string;
+    optimize?: boolean;
+    remotePackage?: string[];
+    remoteUrl?: string;
+    sourceMap?: boolean;
+    strict?: boolean;
+  };
+  webDependencies?: string[];
+}
+
+const defaultConfig: SnowpackConfig = {
+  dedupe: [],
+  options: {
+    clean: false,
+    dest: 'web_modules',
+    exclude: ['**/__tests__/*', '**/*.@(spec|test).@(js|mjs)'],
+    externalPackage: [],
+    nomoduleOutput: 'app.nomodule.js',
+    optimize: false,
+    remoteUrl: 'https://cdn.pika.dev',
+    strict: false,
+  },
+};
+
+const configSchema = {
+  type: 'object',
+  properties: {
+    dedupe: {
+      type: 'array',
+      items: {type: 'string'},
+    },
+    namedExports: {
+      type: 'object',
+      additionalProperties: {type: 'array', items: {type: 'string'}},
+    },
+    options: {
+      type: 'object',
+      properties: {
+        babel: {type: 'boolean'},
+        clean: {type: 'boolean'},
+        dest: {type: 'string'},
+        exclude: {type: 'array', items: {type: 'string'}},
+        externalPackage: {type: 'array', items: {type: 'string'}},
+        include: {type: 'string'},
+        nomodule: {type: 'string'},
+        nomoduleOutput: {type: 'string'},
+        optimize: {type: 'boolean'},
+        remotePackage: {type: 'array', items: {type: 'string'}},
+        remoteUrl: {type: 'string'},
+        sourceMap: {type: 'boolean'},
+        strict: {type: 'boolean'},
+      },
+    },
+    webDependencies: {type: 'array', items: {type: 'string'}},
+  },
+};
+
+export default async function loadConfig(cliFlags?: SnowpackConfig) {
+  const explorerSync = cosmiconfigSync('snowpack');
+  const result = explorerSync.search(); // search for snowpack config
```

Comment:
Okay, this is probably the part of this PR that I care the most about: cosmicconfig's approach to "we'll load your config from anywhere!" makes me very nervous that it will add a lot of maintenance cost, documentation cost and potential bugs going forward. At least they don't do crazy merging like `rc`, but still I'd prefer that we start very simple & locked down here in this first PR and then only add additional config languages/functionality if our users ask for it.

With that in mind, here's what I propose:
```js
import {cosmiconfig, defaultLoaders} from 'consmicconfig';
const explorer = await cosmiconfig('snowpack', {
  // Only support/document these 3 types of config:
  searchPlaces: [
    'package.json', 
    `snowpack.config.js`, 
    `snowpack.config.json`
  ],
  // To start, don't support crawling up the folder tree:
  stopDir: path.resolve(process.cwd(), '../'),
});
```

What do you think? I could see YAML support also being a good fit (`snowpack.config.yaml`) but I'd really like to not support an extension-less `.snowpackrc` file.  Most projects that I know of are moving away from it (Babel, ESLint, etc).

Reply:
Oh yes. I like this. +1 on reducing number of possible configs.

So what would be your recommendation for the default? You’re right—I do see `*.config.js` more and more as the default, but it’d be a bit of cruel irony if the recommended way to configure an ESM bundler would be to use CommonJS!

Reply:
lol I know, hopefully not for too much longer!

I guess I'd rather document everything on the docs site as JSON, just for consistency with what we have so far. And you make a good point about not forcing our users to write CJS :)